main.py

```
@@ -1 +1,57 @@
+        url = self.entry.get().strip()
+        if not url:
+            return
+        threading.Thread(target=self.download_video, args=(url,), daemon=True).start()
+
+    def download_video(self, url: str) -> None:
+        opts = {"outtmpl": "%(title)s.%(ext)s"}
+        try:
+            with YoutubeDL(opts) as ydl:
+                ydl.download([url])
+            self.root.after(0, lambda: messagebox.showinfo("Sucesso", "Download concluído!"))
+        except Exception as exc:
```

Ruff (0.12.2)

Local variable `exc` is assigned to but never used

Remove assignment to unused variable `exc`

(F841)